### PR TITLE
Update Discovery Protocols documentation [8474]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/408/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/412/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/427/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/428/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/429/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/431/)

File: docs/fastdds/discovery/includes/aliases.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.. |DomainId| replace:: :cpp:var:`DomainId <eprosima::fastdds::dds::DomainId_t>`
+.. |DomainParticipantQos| replace:: :cpp:class:`DomainParticipantQos <eprosima::fastdds::dds::DomainParticipantQos>`
+.. |DomainParticipantQosWireProtocol| replace:: :cpp:func:`wire_protocol() <eprosima::fastdds::dds::DomainParticipantQos::wire_protocol()>`
+.. |DomainParticipantQosWireProtocolClass| replace:: :cpp:class:`DomainParticipantQosWireProtocol <eprosima::fastdds::dds::WireProtocolConfigQos>`
```
Comment (EduPonz) [inline docs/fastdds/discovery/includes/aliases.rst:4]:
```suggestion
.. |DomainParticipantQosWireProtocolClass| replace:: :cpp:class:`WireProtocolConfigQos <eprosima::fastdds::dds::WireProtocolConfigQos>`
```

File: docs/fastdds/discovery/server_client.rst [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+^^^^^^^^^^^^
+
+In this architecture there are several key concepts to understand:
+
+- The Server-Client discovery mechanism reuses the RTPS discovery messages structure, as well as the standard DDS
+  |DataWriters| and |DataReaders|.
+
+- Discovery server |DomainParticipants| may be *clients* or *servers*.
+  The only difference between them is how they handle meta-traffic.
+  The user traffic, that is, the traffic among the |DataWriters| and |DataReaders| they create is role-independent.
```
Comment (EduPonz) [inline docs/fastdds/discovery/server_client.rst:34]:
```suggestion
  The user traffic, that is, the traffic among the |DataWriters| and |DataReaders| they create, is role-independent.
```

File: docs/fastdds/discovery/server_client.rst [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    This identifier allows clients to assess whether they are receiving messages from the right server, as each
+    standard RTPS message contains this piece of information.
+
+    The ``GuidPrefix`` is used because the server's IP address may not be a reliable enough server identifier,
+    since several servers can be hosted in the same machine, thus having the same IP, and also because multicast
+    addresses are acceptable addresses.
+
+- *Servers* do not require any beforehand knowledge of their *clients*, but their ``GuidPrefix`` and locator list (where
+  they are listening) must match the one provided to the *clients*.
+  In order to gather *client* discovery information the following handshake strategy is followed:
```
Comment (EduPonz) [inline docs/fastdds/discovery/server_client.rst:54]:
```suggestion
  In order to gather *client* discovery information, the following handshake strategy is followed:
```

File: docs/fastdds/discovery/server_client.rst [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+
+  - *Clients* send discovery messages to the *servers* at regular intervals (ping period) until they receive message
+    reception acknowledgement.
+
+  - *Servers* receive discovery messages from the clients, but they do not start processing them until a time interval
+    has elapsed, which starts at the moment the server is instantiated.
+
+In order to clarify this discovery setup, either on compile time (sources) or runtime (XML files), its explanation is
+divided into two sections: focusing on the main concepts (:ref:`setup by concept <DS_setup_concepts>`), and on
+the main setting structures and XML tags (:ref:`setup by QoS <DS_setup_attributes>`).
```
Comment (EduPonz) [inline docs/fastdds/discovery/server_client.rst:64]:
```suggestion
In order to clarify this discovery setup, either on compile time (sources) or runtime (XML files), this explanation is
divided into two sections: on focusing on the main concepts (:ref:`setup by concept <DS_setup_concepts>`), and another one focusing on
the main setting structures and XML tags (:ref:`setup by QoS <DS_setup_attributes>`).
```

File: code/DDSCodeTester.cpp [hunk 5]
```diff
@@ -510,6 +510,287 @@ void dds_domain_examples()
...
+    }
+
+    /* Custom Callback on_publisher_discovery */
+    virtual void on_publisher_discovery(
+            DomainParticipant* participant,
+            eprosima::fastrtps::rtps::WriterDiscoveryInfo&& info)
+    {
+        (void)participant;
+        switch (info.status){
+            case eprosima::fastrtps::rtps::WriterDiscoveryInfo ::DISCOVERED_WRITER:
```
Comment (EduPonz) [inline code/DDSCodeTester.cpp:571]:
```suggestion
            case eprosima::fastrtps::rtps::WriterDiscoveryInfo::DISCOVERED_WRITER:
```

File: code/DDSCodeTester.cpp [hunk 6]
```diff
@@ -510,6 +510,287 @@ void dds_domain_examples()
...
+                /* Process the case when a new publisher was found in the domain */
+                std::cout << "New DataWriter publishing under topic '" << info.info.topicName() <<
+                    "' of type '" << info.info.typeName() << "' discovered";
+                break;
+            case eprosima::fastrtps::rtps::WriterDiscoveryInfo ::CHANGED_QOS_WRITER:
```
Comment (EduPonz) [inline code/DDSCodeTester.cpp:576]:
```suggestion
            case eprosima::fastrtps::rtps::WriterDiscoveryInfo::CHANGED_QOS_WRITER:
```

File: code/DDSCodeTester.cpp [hunk 7]
```diff
@@ -510,6 +510,287 @@ void dds_domain_examples()
...
+                /* Process the case when a publisher changed its QOS */
+                break;
+            case eprosima::fastrtps::rtps::WriterDiscoveryInfo ::REMOVED_WRITER:
```
Comment (EduPonz) [inline code/DDSCodeTester.cpp:579]:
```suggestion
            case eprosima::fastrtps::rtps::WriterDiscoveryInfo::REMOVED_WRITER:
```

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/432/)

**Pull request merged**